Tests/ArcGISToolkitTests/PreplannedMapModelTests.swift

```diff
@@ -0,0 +1,201 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+import ArcGIS
+@testable import ArcGISToolkit
+
+private extension PreplannedMapAreaProtocol {
+    var packagingStatus: PreplannedMapArea.PackagingStatus? { nil }
+    var title: String { "Mock Preplanned Map Area" }
+    var description: String { "This is the description text" }
+    var thumbnail: ArcGIS.LoadableImage? { nil }
+}
+
+class PreplannedMapModelTests: XCTestCase {
+    private static var sleepNanoseconds: UInt64 { 100_000 }
+    
+    @MainActor
+    func testInitialStatus() {
+        class MockPreplannedMapArea: PreplannedMapAreaProtocol {
+            func retryLoad() async throws { }
+        }
+        
+        let mockArea = MockPreplannedMapArea()
+        let model = PreplannedMapModel(preplannedMapArea: mockArea)
+        guard case .notLoaded = model.status else {
+            XCTFail("PreplannedMapModel initial status is not \".notLoaded\".")
+            return
+        }
+        
+        XCTAssertFalse(model.canDownload)
+    }
+    
+    @MainActor
+    func testLoadingStatus() async throws {
+        class MockPreplannedMapArea: PreplannedMapAreaProtocol {
+            func retryLoad() async throws {
+                // Delay `retryLoad` method to slow down the model's private async
+                // `load` method, so the model status stays at "loading".
+                try await Task.sleep(nanoseconds: 2 * PreplannedMapModelTests.sleepNanoseconds)
+            }
+        }
+        
+        let mockArea = MockPreplannedMapArea()
+        let model = PreplannedMapModel(preplannedMapArea: mockArea)
+        try await Task.sleep(nanoseconds: PreplannedMapModelTests.sleepNanoseconds)
```

Comment:
As discussed, I used `Task.sleep` to wait for the detached `Task` in the initializer. Please let me know if there is a better approach.